src/classes/frModel.cls

```diff
@@ -0,0 +1,71 @@
+public abstract class frModel {
+	protected abstract SObject getObject();
+    public abstract List<frMapping__c> getMappings();
+
+    public virtual void populateFromRequest(Map<String, Object> request) {
+    	Map<String, Schema.SObjectField> fields = frSchemaUtil.getFields(getObject().getSObjectType().getDescribe().getName());
+
+        Map<String, frMapping__c> frNameToSfName = new Map<String, frMapping__c>();
+        for(frMapping__c mapping : getMappings()) {
+            frNameToSfName.put(mapping.fr_Name__c, mapping);
```

Comment:
Actually turns out that it did create the opportunity, but it exposed something a bit weird. The fr_id__c field exists from the old version and in the new version, and the queries to find donors by fr id are all looking at the new version. Is this something we want to do anything about or is this just a "gotta data loader it, fool" type issue to move the funraise ids from the old column to the new one?